### PR TITLE
Support for AutoSuggestBox.QueryIcon

### LINKED ISSUE #5545: AutoSuggestBox always has a Find icon
## Current behavior

`AutoSuggestBox` always uses the `Find` query icon.

## Expected behavior

`AutoSuggestBox` should use the icon user specifies.

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):
**All**

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] macOS
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): fixes #5545

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

Not supported

## What is the new behavior?

Supported

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs [hunk 1]
```diff
@@ -276,6 +272,24 @@ private void OnIsSuggestionListOpenChanged(DependencyPropertyChangedEventArgs e)
...
 
+		private void UpdateQueryButton()
+		{
+			if (_queryButton == null)
+			{
+				return;
+			}
+
+			_queryButton.Content = QueryIcon;
+			if (QueryIcon == null)
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.cs:283]: Use ternary

**Review by jeromelaban — APPROVED** [review]

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/MeasuredBitmapIcon.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using Windows.UI.Xaml.Controls;
+
+namespace UITests.Windows_UI_Xaml_Controls.AutoSuggestBoxTests
+{
+    public partial class MeasuredBitmapIcon : BitmapIcon
+    {
+		protected override Size ArrangeOverride(Size finalSize)
+		{
+			var size = base.ArrangeOverride(finalSize);
+			global::System.Diagnostics.Debug.WriteLine("Arrange final size " + finalSize + ", result size " + size);
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/AutoSuggestBoxTests/MeasuredBitmapIcon.cs:16]: Codacy missed something here...

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/BitmapIconTests/BitmapIcon_Generic.xaml.cs [hunk 3]
```diff
@@ -15,11 +15,9 @@
 using Windows.UI.Xaml.Media;
 using Windows.UI.Xaml.Navigation;
 
-// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
-
 namespace UITests.Shared.Windows_UI_Xaml_Controls.BitmapIconTests
 {
-	[SampleControlInfo("BitmapIcon", nameof(BitmapIcon_Generic))]
+	[SampleControlInfo("IconElement", nameof(BitmapIcon_Generic))]
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/BitmapIconTests/BitmapIcon_Generic.xaml.cs:20]:
```suggestion
	[Sample("IconElement")]
```

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/FontIcon/FontIconControlTest.xaml.cs [hunk 4]
```diff
@@ -3,7 +3,7 @@
 
 namespace Uno.UI.Samples.Content.UITests.FontIconTests
 {
-	[SampleControlInfo("FontIcon", "FontIconControlTest")]
+	[SampleControlInfo("IconElement", "FontIconControlTest")]
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Controls/FontIcon/FontIconControlTest.xaml.cs:6]:
```suggestion
	[Sample("IconElement")]
```

File: src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Shapes/PathTestsControl/PathIcon.xaml.cs [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 namespace SamplesApp.Windows_UI_Xaml_Shapes.PathTestsControl
 {
-	[SampleControlInfo("PathTestsControl", "PathIcon")]
+	[SampleControlInfo("IconElement", "PathIcon")]
```
Comment (carldebilly) [inline src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Shapes/PathTestsControl/PathIcon.xaml.cs:6]:
```suggestion
	[Sample("IconElement")]
```

**Review by carldebilly — APPROVED** [review]

**Pull request merged**